### PR TITLE
Make Chef CI use existing zzz_generated and complete Chef CD for existing platforms

### PR DESCRIPTION
**Pull request by aBozowski**
#### Problem
What is being fixed?  Examples:
* Chef CI has its own regen strategy
* Completing Chef CD for the platforms that are in CI.

#### Change overview
* Update regen_all and the chef CI job to standardize zzz files
* Adds bundle functions to filter build outputs to archives.
* Adds error handling to `--build_all`.
* Update `--generate_zzz` and `--validate_zzz` for `.matter` files.

#### Testing
How was this tested? (at least one bullet point required)
* Tested with vscode container mounting CHIP dir and a large set of device types.
* Existing CI jobs cover the new paths.

### CODE REVIEW

File: examples/chef/cicd_config.json [hunk 1]
```diff
@@ -0,0 +1,8 @@
+{
+    "ci_allow_list": ["lighting-app"],
+    "cd_platforms": {
+        "linux": "linux_x86",
+        "esp32": "esp32-m5stack",
```
[the remark this answers is not in the document]
Comment (aBozowski) [reply examples/chef/cicd_config.json:5]: Updating the labels is fine.

> Also note that these platform names may have additional suffixes, but that would depend on build flags.

Are you saying this is a current requirement, or is this just a note for later? If so, what property of the device file dictates this.

File: examples/chef/chef.py [hunk 2]
```diff
@@ -168,11 +169,132 @@ def flush_print(
...
+        device_name: The device to bundle.
+    """
+    esp_root = os.path.join(_CHEF_SCRIPT_PATH,
+                            "esp32",
+                            "build")
+    manifest_file = os.path.join(esp_root,
+                                 "chip-shell.flashbundle.txt")
+    with open(manifest_file) as manifest:
+        for item in manifest:
+            item = item.replace("\n", "")
```
[the remark this answers is not in the document]
Comment (aBozowski) [reply examples/chef/chef.py:288]: Whitespace is not a concern but I can change it.

File: examples/chef/cicd_config.json (hunk 1 above)
[the remark this answers is not in the document]
Comment (aBozowski) [reply examples/chef/cicd_config.json]: Thank you I have updated the labels for now.

File: examples/chef/chef.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (aBozowski) [reply examples/chef/chef.py]: Changed thanks.

File: examples/chef/chef.py [hunk 3]
```diff
@@ -168,11 +169,132 @@ def flush_print(
...
+            if os.sep in item:
+                new_dir = item[:item.rindex(os.sep)]
+                new_dir = os.path.join(_CD_STAGING_DIR, new_dir)
+                os.makedirs(new_dir, exist_ok=True)
+            src_item = os.path.join(esp_root, item)
+            dest_item = os.path.join(_CD_STAGING_DIR, item)
```
[the remark this answers is not in the document]
Comment (aBozowski) [reply examples/chef/chef.py]: This is better; changed thanks.

File: examples/chef/chef.py [hunk 4]
```diff
@@ -168,11 +169,132 @@ def flush_print(
...
+    matter_file = f"{device_name}.matter"
+    zap_file = os.path.join(_DEVICE_FOLDER, f"{device_name}.zap")
+    flush_print(f"Bundling {platform}", with_border=True)
+    flush_print(f"Cleaning {_CD_STAGING_DIR}")
+    shutil.rmtree(_CD_STAGING_DIR, ignore_errors=True)
+    os.mkdir(_CD_STAGING_DIR)
+    flush_print(f"Checking for {bundler_name}")
+    if bundler_name in globals():
+        flush_print(f"Found {bundler_name}")
+        globals()[bundler_name](device_name)
```
[the remark this answers is not in the document]
Comment (aBozowski) [reply examples/chef/chef.py]: I agree `globals()`was lazy and unsafe, but I feel having to keep a mapping is redundant. I changed this to find the function specifically in chef module. Bundler could be a module or class of its own in the future.

**Review by cpagravel — APPROVED** [review]
One comment about power feature usage (e.g. getattr). Would prefer something a bit simpler. Otherwise LGTM.

File: examples/chef/chef.py [hunk 5]
```diff
@@ -168,11 +169,131 @@ def flush_print(
...
+    flush_print(f"Bundling {platform}", with_border=True)
+    flush_print(f"Cleaning {_CD_STAGING_DIR}")
+    shutil.rmtree(_CD_STAGING_DIR, ignore_errors=True)
+    os.mkdir(_CD_STAGING_DIR)
+    flush_print(f"Checking for {bundler_name}")
+    chef_module = sys.modules[__name__]
+    if hasattr(chef_module, bundler_name):
+        flush_print(f"Found {bundler_name}")
+        bundler = getattr(chef_module, bundler_name)
+        bundler(device_name)
```
[the remark this answers is not in the document]
Comment (aBozowski) [reply examples/chef/chef.py]: I am not fond of the per-platform if/elif branching already present in the rest of the module, but I changed it as suggested for style guide; can just wait for some larger refactor which encapsulates each platform better. Thanks again for review!

**Review by cpagravel — APPROVED** [review]

**Review by tcarmelveilleux — APPROVED** [review]
Example-only changes

**Pull request merged**